airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java

```diff
@@ -857,6 +867,168 @@ public void testSyncAfterUpgradeToPerStreamStateWithNoNewData(final TestInfo tes
+        .schedule(connection.getSchedule())
+        .sourceCatalogId(connection.getSourceCatalogId())
+        .status(connection.getStatus())
+        .resourceRequirements(connection.getResourceRequirements())
+        .withRefreshedCatalog(true);
+    final WebBackendConnectionRead connectionUpdateRead = webBackendApi.webBackendUpdateConnection(update);
+
+    LOGGER.info("Inspecting Destination DB after the update request, tables should be empty");
+    destDb.query(ctx -> {
+      prettyPrintTables(ctx, outputPrefix + sourceTable1, outputPrefix + sourceTable2);
+      return null;
+    });
+    final ConnectionState postResetState =
+        apiClient.getConnectionApi().getState(new ConnectionIdRequestBody().connectionId(connection.getConnectionId()));
+    LOGGER.info("ConnectionState after the update request: {}", postResetState.toString());
+
+    // Wait until the sync from the UpdateConnection is finished
+    final JobRead syncFromTheUpdate = waitUntilTheNextJobIsStarted(connection.getConnectionId());
+    LOGGER.info("Generated SyncJob config: {}", syncFromTheUpdate.toString());
+    waitForSuccessfulJob(apiClient.getJobsApi(), syncFromTheUpdate);
+
+    final ConnectionState postUpdateState =
+        apiClient.getConnectionApi().getState(new ConnectionIdRequestBody().connectionId(connection.getConnectionId()));
+    LOGGER.info("ConnectionState after the final sync: {}", postUpdateState.toString());
+
+    LOGGER.info("Inspecting DBs After the final sync");
+    sourceDb.query(ctx -> {
+      prettyPrintTables(ctx, sourceTable1, sourceTable2, sourceTable3);
+      return null;
+    });
+    destDb.query(ctx -> {
+      prettyPrintTables(ctx, outputPrefix + sourceTable1, outputPrefix + sourceTable2, outputPrefix + sourceTable3);
+      return null;
+    });
+
+    testHarness.assertSourceAndDestinationDbInSync(false);
+  }
+
+  private void prettyPrintTables(final DSLContext ctx, final String... tables) {
+    for (final String table : tables) {
+      LOGGER.info("select * from {}", table);
+      Arrays.stream(ctx.selectFrom(table)
+          .fetch()
+          .toString()
+          .split("\\n")).forEach(LOGGER::info);
+    }
+  }
+
+  private JobRead getMostRecentSyncJobId(UUID connectionId) throws Exception {
+    return apiClient.getJobsApi()
+        .listJobsFor(new JobListRequestBody().configId(connectionId.toString()).configTypes(List.of(JobConfigType.SYNC)))
+        .getJobs()
+        .stream().findFirst().map(JobWithAttemptsRead::getJob).orElseThrow();
+  }
+
+  private JobRead waitUntilTheNextJobIsStarted(final UUID connectionId) throws Exception {
+    final JobRead lastJob = getMostRecentSyncJobId(connectionId);
+    if (lastJob.getStatus() != JobStatus.SUCCEEDED) {
+      return lastJob;
+    }
+
+    JobRead mostRecentSyncJob = getMostRecentSyncJobId(connectionId);
+    while (mostRecentSyncJob.getId().equals(lastJob.getId())) {
+      Thread.sleep(Duration.ofSeconds(10).toMillis());
```

Comment:
should we decrease that, 10 seconds seems big.

Reply:
Lowered it to 1 second. In practice we are still looking at a fairly slow operation, I don't think it makes sense to lower it below a second.